test/core/image/image.cpp

```diff
@@ -33,3 +33,26 @@ BOOST_AUTO_TEST_CASE_TEMPLATE(constructor_with_dimensions_pixel, Image, fixture:
         }
     }
 }
+
+BOOST_AUTO_TEST_CASE_TEMPLATE(move_constructor, Image, fixture::image_types)
+{
+    gil::point_t const dimensions{256, 128};
+    {
+        Image image(fixture::create_image<Image>(dimensions.x, dimensions.y, 0));
+
+        Image image2(std::move(image));
+        BOOST_CHECK_EQUAL(image2.dimensions(), dimensions);
+        BOOST_CHECK_EQUAL(image.dimensions(), gil::point_t{});
+    }
+}
+
+BOOST_AUTO_TEST_CASE_TEMPLATE(move_assignement, Image, fixture::image_types)
+{
+    gil::point_t const dimensions{256, 128};
+    {
+        Image image = fixture::create_image<Image>(dimensions.x, dimensions.y, 0);
+        Image image2 = std::move(image);
```

Comment:
This does not kick the move assignment.
I'd also generate a distinct `image2` that will be deallocated and cleared as prep to be come a moved-to object:

```cpp
image_t image2 = fixture::create_image<image_t>(dimensions.x * 2, dimensions.y * 2, 1);
image2 = std::move(image);
```

Reply:
My bad, this is indeed a construction. I'll fix this. I'm not sure what is the right way to test move semantic...

Reply:
> I'm not sure what is the right way to test move semantic

I'm not 100% either, but looks like testing states before/after of moved-from/to is an obvious way that should give at least basic indication things work.